src/applications/Applications.ExtractImages/ExtractionMessageSender.cs

```diff
@@ -118,7 +121,7 @@ public void SendMessages(string absoluteExtractionDir, ExtractionKey extractionK
 
             if (_nonInteractive)
             {
-                SendMessagesImpl(absoluteExtractionDir, ermList, erim);
+                BeginExtraction(jobId, ermList, erim);
```

Comment:
The prefix `Begin` usually indicates a method is async and non blocking (e.g. `BeginRead`, `BeginInvoke` etc).  I would rename this something else for better readability/maintainability later on but its up to you.

Reply:
Good point - renamed now.